lib/command/Command.js

```diff
@@ -156,13 +156,20 @@ class Command {
         return true;
     }
 
-    process(args, msg) {
+    process(args, msg, client) {
         if(!this.permissionCheck(msg)) {
             if(this.permissionMessage) {
                 msg.channel.createMessage(this.permissionMessage);
             }
             return;
         }
+        if (this.deleteCommand && msg.channel.guild && msg.channel.permissionsOf(client.user.id).has('manageMessages')) {
+            let originalExec = this.execute;
+            this.execute = (msg, args) => {
+                client.deleteMessage(msg.channel.id, msg.id);
```

Comment:
This can be done directly through the Message object, removing the need to pass client